dlt/common/destination/reference.py

```diff
@@ -453,9 +450,9 @@ def capabilities(self) -> DestinationCapabilitiesContext:
         ...
 
     @property
-    def name(self) -> str:
+    def destination_name(self) -> str:
         """The destination name will either be explicitely set while creating the destination or will be taken from the type"""
-        return self.config_params.get("name") or self.destination_type
+        return self.config_params.get("destination_name") or self.destination_type
 
     @property
     def destination_type(self) -> str:
```

Comment:
the line below returns wrong type

Reply:
maybe let's make it to return Type? then you'll catch all the bugs

Reply:
I am not sure what you mean. We already have the type, no? I have changed it that it will return the full path to the destination module and added / updated tests for this.